### ISSUE TITLE
App Crash when use the GCSVideoView to load the local video?

### ISSUE THREAD

**Issue by lianguowu** [googlevr/gvr-ios-sdk#13]
Assertion failure in -[GCSSphericalMetadataExtractor intFromBytes:withLength:], googlemac/iPhone/CardboardSDK/Source/GCSSphericalMetadataExtractor.mm:201
2016-04-21 13:09:18.633 P2[5984:1078876] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'data == nil || length > 8'
*** First throw call stack:
(0x1835cee38 0x182c33f80 0x1835ced08 0x183f54124 0x1000d3fc4 0x1000d3978 0x1000d3660 0x1000d6398 0x183d49694 0x183d49624 0x183d49798 0x183d1ce78 0x183bac93c 0x183ca32a8 0x1007a1a3c 0x1007ab1e0 0x183b9bc6c 0x1834ac73c 0x183b9bb50 0x183b9ba10 0x183b9b83c 0x183585124 0x183584bb8 0x1835828b8 0x1834acd10 0x184d94088 0x188781f70 0x1000a58a8 0x18304a8b8)
libc++abi.dylib: terminating with uncaught exception of type NSException

**Comment by nathanmartz** [comment]
Are you using 0.7.0 or 0.7.1?

**Comment by lianguowu** [comment]
yes i am using 0.7.1

**Comment by seanadkinson** [comment]
I am seeing this same issue.  Seems to work with some videos, but not others.  Perhaps there is some aspect ratio assertions or something?  Is this code not open-source?

**Comment by seanadkinson** [comment]
I just updated to 0.7.2, and still see this issue, btw.

**Comment by sanjayc77** [comment]
Yeah, it was too late to fix it in 0.7.2. We are looking into it.

**Comment by seanadkinson** [comment]
@sanjayc77 FYI, I was able to get my video working by changing the resolution and aspect ratio.

Here is the `ffprobe` of the original non-working video:
```
$ ffprobe garden_falls_orig.mp4 
ffprobe version 2.8.6 Copyright (c) 2007-2016 the FFmpeg developers
  built with Apple LLVM version 7.0.2 (clang-700.1.81)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/2.8.6 --enable-shared --enable-pthreads --enable-gpl --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-opencl --enable-libx264 --enable-libmp3lame --enable-libvo-aacenc --enable-libxvid --enable-vda
  libavutil      54. 31.100 / 54. 31.100
  libavcodec     56. 60.100 / 56. 60.100
  libavformat    56. 40.101 / 56. 40.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 40.101 /  5. 40.101
  libavresample   2.  1.  0 /  2.  1.  0
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  2.101 /  1.  2.101
  libpostproc    53.  3.100 / 53.  3.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'garden_falls_orig.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: mp42mp41
    creation_time   : 2016-02-03 14:50:39
  Duration: 00:00:42.09, start: 0.000000, bitrate: 12594 kb/s
    Stream #0:0(eng): Video: h264 (Main) (avc1 / 0x31637661), yuv420p(tv), 3840x1920 [SAR 8:9 DAR 16:9], 12280 kb/s, SAR 3413:3840 DAR 3413:1920, 24 fps, 24 tbr, 24k tbn, 48 tbc (default)
    Metadata:
      creation_time   : 2016-02-03 14:50:39
      handler_name    : Alias Data Handler
      encoder         : AVC Coding
    Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 317 kb/s (default)
    Metadata:
      creation_time   : 2016-02-03 14:50:39
      handler_name    : Alias Data Handler
```

And here is the one that works:
```
$ ffprobe garden_falls.mp4 
ffprobe version 2.8.6 Copyright (c) 2007-2016 the FFmpeg developers
  built with Apple LLVM version 7.0.2 (clang-700.1.81)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/2.8.6 --enable-shared --enable-pthreads --enable-gpl --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-opencl --enable-libx264 --enable-libmp3lame --enable-libvo-aacenc --enable-libxvid --enable-vda
  libavutil      54. 31.100 / 54. 31.100
  libavcodec     56. 60.100 / 56. 60.100
  libavformat    56. 40.101 / 56. 40.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 40.101 /  5. 40.101
  libavresample   2.  1.  0 /  2.  1.  0
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  2.101 /  1.  2.101
  libpostproc    53.  3.100 / 53.  3.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'garden_falls.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf56.40.101
  Duration: 00:00:42.09, start: 0.000000, bitrate: 28607 kb/s
    Stream #0:0(eng): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 4096x2048 [SAR 1:1 DAR 2:1], 28289 kb/s, 24 fps, 24 tbr, 12288 tbn, 48 tbc (default)
    Metadata:
      handler_name    : VideoHandler
    Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 317 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
```

Here is the command I used to convert it:
```
$ ffmpeg -i garden_falls_orig.mp4 -vf scale=4096:2048,setsar=1:1 -c:v libx264 -crf 20 -c:a copy garden_falls.mp4 
```

**Comment by fxd** [comment]
could you email you converted video to me,thanks

**Comment by sanjayc77** [comment]
This has been fixed in GVRSDK 0.8.0.
